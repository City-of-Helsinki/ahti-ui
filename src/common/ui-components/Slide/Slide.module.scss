.container {
  height: 300px;
  margin: 0 1rem 0 0;
  background-color: pink;
  background-size: cover;
  position: relative;
  padding: 2rem;

  h3 {
    color: white;
    position: absolute;
    bottom: 2rem;
  }
}
