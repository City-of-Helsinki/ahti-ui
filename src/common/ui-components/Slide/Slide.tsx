import React from 'react';
import classNames from 'classnames/bind';

import styles from './Slide.module.scss';

export interface SlideProps {
  children: React.ReactNode;
}

const Slide = (props: SlideProps) => {
  return <div className={classNames(styles.container)}>{props.children}</div>;
};

export default Slide;
