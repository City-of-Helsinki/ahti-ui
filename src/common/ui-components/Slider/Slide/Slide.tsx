import React from 'react';
import classNames from 'classnames';

import styles from './Slide.module.scss';

export interface SlideProps {
  readonly imageUrl: string;
  readonly className?: string;
  onClick?(): void;
}

const Slide: React.FC<SlideProps> = ({
  imageUrl,
  className,
  children,
  onClick,
}) => {
  return (
    <div
      className={classNames(className, styles.slide)}
      style={{ backgroundImage: `url(${imageUrl})`, backgroundSize: 'cover' }}
      onClick={() => onClick && onClick()}
      role={'button'}
    >
      {children}
    </div>
  );
};

export default Slide;
