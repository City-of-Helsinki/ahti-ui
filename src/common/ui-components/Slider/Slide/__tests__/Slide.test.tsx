import React from 'react';
import { shallow } from 'enzyme';

import Slide from '../Slide';

describe('Slide', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Slide imageUrl={''} />);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
