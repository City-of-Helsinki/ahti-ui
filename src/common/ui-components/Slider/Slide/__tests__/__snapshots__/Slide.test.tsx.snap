// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide renders without crashing 1`] = `"<div class=\\"slide\\" style=\\"background-image:url();background-size:cover\\" role=\\"button\\"></div>"`;
